### PR TITLE
Rewrite tests for simple stage tester

### PR DESCRIPTION
**Pull request by timohanke**
The PR is converting all tests from variable tester to simple stage tester.

I think it makes the tests clearer. The advantage is that simple stage tester enforces us to write clearer tests.

Some example I noticed in the tests with variable tester:

Sometimes the set() was missing in the test block. So to understand the test the reader has to search backwards to the last occurrence of set to find the value. The simple stage tester does not allow that. If the response isn't staged it will fail with "pop from empty queue".

Sometimes there were unused calls to set(). The variable was set but the value was never read. The simple stage tester does not allow that. If a response is staged and not read then the subsequent test will get a wrong response and will likely fail.

Sometimes we want to assert that no call was made. With the variable tester we have to introduce the counter value in the mock_ledger and assert that the value doesn't increase. With the simple stage tester we can simply not stage a response and thereby implicitly make that assertion. Because if the code does make a call (against our assertion) then the test will catch that with "pop from empty queue".

One time there was even no set() call at all (not only an older one) and the call responded with the default value defined in the class constructor.

Overall I think the variable tester can hide test bugs and the strictness simple stage tester is better.

### CODE REVIEW

File: test/consolidation.test.mo [hunk 1]
```diff
@@ -173,16 +163,21 @@ do {
...
     #issued(+1),
     #newDeposit(8),
   ]);
 
-  // wait for consolidation
-  await async {};
+  // Wait for consolidation
+  //   1. wait for transfer() response to be ready 
+  //   similar: await* mock_ledger.transfer_.wait(i);
+  while (mock_ledger.transfer_.state(i) != #ready) await async {};
```
Comment (AStepanov25) [inline test/consolidation.test.mo]: Why not `await* mock_ledger.transfer_.wait(i);` ?

Comment (AStepanov25) [reply test/consolidation.test.mo]: BTW, we don't need util/mock.mo now.

Comment (timohanke) [reply test/consolidation.test.mo]: Ok, both done.

Would be nice to get rid of the step 2. `await async {}` but doesn't seem to be possible.

**Review by AStepanov25 — APPROVED** [review]

**Comment by timohanke** [comment]
Closing this because the same tests have been added to the refactored branch in PR #21 instead.

**Pull request closed without merging**